types/index.d.ts

```diff
@@ -1430,7 +1440,7 @@ export declare namespace Knex {
         // causes rest of the keys to be dropped and only first to be considered
         [K in keyof TAliases]?: K extends keyof TRecord ?
           TRecord[K] :
-          TValue
+          UnknownToAny<ResolveTableType<TRecord[TAliases[K]]>>
```

Comment:
The test currently fails with `Type TAliases[K] cannot be used to index type TRecord. ` Does anyone know what could be the best way to handle this case?
The idea is 
```typescript
  // $ExpectType ({ dep: number; } & { a?: number; } & { b?: number; })[]
  await knexInstance<User>('users')
    .select({dep: 'departmentId'})
    .min({a: 'age'} as const)
    .max({b: 'age'} as const);
```